### PR TITLE
Docs/robots

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

fix canonical, @aribornstein 

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by SeanNaren — APPROVED** [review]

File: docs/source/conf.py [hunk 1]
```diff
@@ -178,8 +178,8 @@ def _transform_changelog(path_in: str, path_out: str) -> None:
 # documentation.
 
 html_theme_options = {
-    'pytorch_project': info.__homepage__,
-    'canonical_url': info.__homepage__,
+    'pytorch_project': 'https://pytorchlightning.ai',
```
Comment (carmocca) [inline docs/source/conf.py:181]: why not keep `info.__homepage__` and update the link there?

File: docs/source/conf.py [hunk 2]
```diff
@@ -178,8 +178,8 @@ def _transform_changelog(path_in: str, path_out: str) -> None:
...
+    'canonical_url': 'https://pytorch-lightning.readthedocs.io/en/stable/',
```
Comment (carmocca) [inline docs/source/conf.py:182]: I thought we wanted to have `latest` as the default

Comment (Borda) [reply docs/source/conf.py:182]: we shall refer stable version not develop
cc: @aribornstein

File: docs/source/conf.py (hunk 1 above)
Comment (Borda) [reply docs/source/conf.py:181]: some SEO :]
cc: @aribornstein

**Review by s-rog — APPROVED** [review]

File: docs/source/conf.py (hunk 1 above)
Comment (aribornstein) [reply docs/source/conf.py:181]: This points to the github and we want these urls to point to the site where the docs are hosted.

File: docs/source/conf.py (hunk 2 above)
Comment (aribornstein) [reply docs/source/conf.py]: For people looking for the docs as reference I think it’s best for us to point to the stable version as latest is often under development.

Comment (aribornstein) [reply docs/source/conf.py]: Is there a reason you prefer latest to stable?

Comment (Borda) [reply docs/source/conf.py]: I would use stable too, as the stable is the version on pypi so there is less confusion between the installed and documented version

**Review by awaelchli — APPROVED** [review]

**Pull request merged**